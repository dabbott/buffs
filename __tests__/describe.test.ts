import fs from 'fs'
import path from 'path'

import {
  describe as describeFS,
  describeComparison,
  createFs,
} from '../src/index'

const mocksPath = path.join(__dirname, 'mocks')

describe('Describe', () => {
  test('describes a fs', () => {
    const description = describeFS(fs, path.join(mocksPath, 'describe'))

    expect(description).toMatchSnapshot()
  })

  describe('describeComparison', () => {
    const { volume, fs: target } = createFs()

    volume.mkdirpSync(path.join(mocksPath, 'describe', 'b'))
    target.writeFileSync(
      path.join(mocksPath, 'describe', 'b', 'b.txt'),
      'something'
    )

    test('describes a colorized fs comparison', () => {
      const description = describeComparison(
        fs,
        target,
        path.join(mocksPath, 'describe'),
        { colorize: true }
      )

      expect(description).toMatchInlineSnapshot(`
        "[90mdescribe[39m
        â”œâ”€â”€ [32ma / a.txt[39m
        â””â”€â”€ [90mb[39m
            â”œâ”€â”€ [33mb.txt[39m
            â”œâ”€â”€ [32mc[39m
            â”‚Â Â  â”œâ”€â”€ [32mc.txt[39m
            â”‚Â Â  â”œâ”€â”€ [32md / d.txt[39m
            â”‚Â Â  â””â”€â”€ [32me.txt[39m
            â””â”€â”€ [32mf.txt[39m"
      `)
    })

    test('describes an uncolored fs comparison', () => {
      const description = describeComparison(
        fs,
        target,
        path.join(mocksPath, 'describe')
      )

      expect(description).toMatchSnapshot()
    })
  })
})
