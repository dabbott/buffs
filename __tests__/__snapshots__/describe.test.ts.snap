// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Describe describeComparison describes an uncolored fs comparison 1`] = `
"describe[N]
├── a / a.txt[A]
└── b[N]
    ├── b.txt[M]
    ├── c[A]
    │   ├── c.txt[A]
    │   ├── d / d.txt[A]
    │   └── e.txt[A]
    └── f.txt[A]"
`;

exports[`Describe describes a fs 1`] = `
"describe
├── a / a.txt
└── b
    ├── b.txt
    ├── c
    │   ├── c.txt
    │   ├── d / d.txt
    │   └── e.txt
    └── f.txt"
`;
